mteb/abstasks/AbsTaskRegression.py

```diff
@@ -0,0 +1,269 @@
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import datasets
+import numpy as np
+import pandas as pd
+from sklearn.base import BaseEstimator
+from sklearn.linear_model import LinearRegression
+
+from mteb.abstasks.AbsTask import AbsTask
+from mteb.abstasks.TaskMetadata import DescriptiveStatistics, HFSubset
+from mteb.encoder_interface import Encoder
+from mteb.evaluation.evaluators.RegressionEvaluator import LinearRegressionEvaluator
+from mteb.load_results.task_results import ScoresDict
+
+logger = logging.getLogger(__name__)
+
+
+class RegressionDescriptiveStatistics(DescriptiveStatistics):
+    """Descriptive statistics for Regression
+
+    Attributes:
+      num_samples: number of samples in the dataset.
+      number_of_characters: Total number of symbols in the dataset.
+      num_texts_in_train: Number of texts in the train split
+
+      min_text_length: Minimum length of text
+      average_text_length: Average length of text
+      max_text_length: Maximum length of text
+      unique_text: Number of unique texts
+
+      min_value: Minimum of the target variable
+      average_value: Average of the target variable
+      max_value: Maximum of the target variable
+    """
+
+    num_samples: int
+    number_of_characters: int
+    num_texts_in_train: int | None
+
+    min_text_length: int
+    average_text_length: float
+    max_text_length: int
+    unique_text: int
+
+    min_value: float
+    average_value: float
+    max_value: float
+
+
+class AbsTaskRegression(AbsTask):
```

Comment:
Assuming it only worsens for text regression let us rename it to `AbsTaskTextRegression` (for v2 compatibility)